locust/shape.py

```diff
@@ -1,11 +1,14 @@
 from __future__ import annotations
+
 import time
-from typing import ClassVar
 from abc import ABCMeta, abstractmethod
+from typing import TYPE_CHECKING, ClassVar
 
-from . import User
 from .runners import Runner
 
+if TYPE_CHECKING:
```

Comment:
is this needed, or just a remnant of the first solution?

Reply:
Needed, see this [comment](https://github.com/locustio/locust/pull/2556#issuecomment-1902687243).